src-plugins/itkDataImage/writers/itkDicomDataImageWriter.cpp

```diff
@@ -0,0 +1,492 @@
+  the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+  PURPOSE.
+
+=========================================================================*/
+
+#include <itkDicomDataImageWriter.h>
+
+#include <medAbstractData.h>
+#include <medAbstractDataFactory.h>
+
+
+#include <itkExtractImageFilter.h>
+#include <itkGDCMImageIO.h>
+#include <itkImage.h>
+#include <itkImageFileWriter.h>
+#include <itkMetaDataObject.h>
+
+#include <gdcmUIDGenerator.h>
+
+#include <string>
+
+static QString s_identifier() {
+    return "itkDicomDataImageWriter";
+}
+
+static QStringList s_handled() {
+    return QStringList()  << "itkDataImageChar3"
+                          << "itkDataImageUChar3"
+                          << "itkDataImageShort3"
+                          << "itkDataImageUShort3"
+                          << "itkDataImageInt3"
+                          << "itkDataImageUInt3"
+                          << "itkDataImageLong3"
+                          << "itkDataImageULong3"
+                          << "itkDataImageFloat3"
+                          << "itkDataImageDouble3";
+}
+
+itkDicomDataImageWriter::itkDicomDataImageWriter(): itkDataImageWriterBase() {
+    this->io = itk::GDCMImageIO::New();
+}
+
+itkDicomDataImageWriter::~itkDicomDataImageWriter()
+{
+}
+
+QStringList itkDicomDataImageWriter::handled() const
+{
+    return s_handled();
+}
+
+QStringList itkDicomDataImageWriter::supportedFileExtensions() const
+{
+    return QStringList() << ".dcm";
+}
+
+bool itkDicomDataImageWriter::registered()
+{
+    return medAbstractDataFactory::instance()->registerDataWriterType(s_identifier(), s_handled(), create);
+}
+
+QString itkDicomDataImageWriter::identifier() const
+{
+    return s_identifier();
+}
+
+QString itkDicomDataImageWriter::description() const
+{
+    return "Dicom image exporter";
+}
+
+QString itkDicomDataImageWriter::sopClassUID(QString modality)
+{
+    if( modality == QString("CT"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.2";
+    }
+    else if( modality == QString("MR"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.4";
+    }
+    else if( modality == QString("CR"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.1";
+    }
+    else if( modality == QString("MN"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.20";
+    }
+    else if( modality == QString("US"))
+    {
+        return "1.2.840.10008.5.1.4.1.1.6.1";
+    }
+    else
+    {
+        qDebug( "Unknown modality. Using CT Image Storage SOP class UID.");
+        return "1.2.840.10008.5.1.4.1.1.2";
+    }
+}
+
+template <class PixelType> bool itkDicomDataImageWriter::writeDicom(const QString &path)
+{
+    //Trick to use QString with accent
+    QTextCodec::setCodecForCStrings(QTextCodec::codecForName("UTF-8"));
+    //Useful when writting Image position and orientiation matrice
+    setlocale(LC_NUMERIC, "C");
+    QLocale::setDefault(QLocale("C"));
+
+    QString filePath = path.left(path.length() - 4);
```

Comment:
the file path can be get from the `QFileInfo fi(path);` just under

Reply:
To get what I want I have to do QString filePath =  fi.path() + "/" + fi.baseName() which is not better than  QString filePath = path.left(path.length() - 4);